src/gui/guiInventoryList.cpp

```diff
@@ -85,6 +85,11 @@ void GUIInventoryList::draw()
 		v2s32 p((i % m_geom.X) * m_slot_spacing.X,
 				(i / m_geom.X) * m_slot_spacing.Y);
 		core::rect<s32> rect = imgrect + base_pos + p;
+
+		if (!isNotClipped() && Parent &&
+				!Parent->getAbsoluteClippingRect().isRectCollided(rect))
```

Comment:
shouldn't the element have its own clipping rect?
I can see `Parent->getAbsoluteClippingRect` being used below too but it doesn't sound correct at first glance

Reply:
I think these should be about equivalent. `getAbsoluteClippingRect()` is essentially just the intersection of the parent clipping rect and a bounding box of all our contents. The bounding box of our contents necessarily contains these `rect`s. It's equivalent to check whether a slot intersects just the parent, or whether it intersect the intersection of the parent and the bounding box of all slots. With an exception for noclip of course.

But it's definitely nicer if I just use `getAbsoluteClippingRect()` so I'll do just that.